engine/access/rpc/backend/backend_stream_transactions.go

```diff
@@ -0,0 +1,182 @@
+		}
+
+		if txInfo.blockWithTx == nil {
+			// Check if block contains transaction.
+			txInfo.blockWithTx, err = b.searchForTransactionBlock(height, txInfo)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// Find the transaction status.
+		var txStatus flow.TransactionStatus
+		if txInfo.blockWithTx == nil {
+			txStatus, err = b.txLocalDataProvider.DeriveUnknownTransactionStatus(txInfo.txReferenceBlockID)
+		} else {
+			blockID := txInfo.blockWithTx.ID()
+
+			if !txInfo.txExecuted {
+				// Check if transaction was executed.
+				txInfo.txExecuted, err = b.searchForExecutionResult(blockID)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			txStatus, err = b.txLocalDataProvider.DeriveTransactionStatus(blockID, txInfo.blockWithTx.Height, txInfo.txExecuted)
+		}
+		if err != nil {
+			if !errors.Is(err, state.ErrUnknownSnapshotReference) {
+				irrecoverable.Throw(ctx, err)
+			}
+			return nil, rpc.ConvertStorageError(err)
+		}
+
+		messageIndex := txInfo.messageIndex.Value()
+		if ok := txInfo.messageIndex.Set(messageIndex + 1); !ok {
+			return nil, status.Errorf(codes.Internal, "the message index has already been incremented to %d", txInfo.messageIndex.Value())
+		}
+
+		return &convert.TransactionSubscribeInfo{
+			ID:           txInfo.txID,
+			Status:       txStatus,
+			MessageIndex: messageIndex,
+		}, nil
+	}
+}
+
+// searchForTransactionBlock searches for the block containing the specified transaction.
+// It retrieves the block at the given height and checks if the transaction is included in that block.
+// Expected errors:
+// - codes.Internal when unable to retrieve the block or collection ID
+func (b *backendSubscribeTransactions) searchForTransactionBlock(
+	height uint64,
+	txInfo *TransactionSubscriptionMetadata,
+) (*flow.Header, error) {
+	block, err := b.txLocalDataProvider.blocks.ByHeight(height)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not get block %d: %v", height, err)
+	}
+
+	collectionID, err := b.txLocalDataProvider.LookupCollectionIDInBlock(block, txInfo.txID)
+	if err != nil {
+		if status.Code(err) != codes.NotFound {
+			return nil, status.Errorf(codes.Internal, "could not find transaction in block: %v", err)
+		}
+	}
+
+	if collectionID != flow.ZeroID {
+		return block.Header, nil
+	}
+
+	return nil, nil
+}
+
+// searchForExecutionResult searches for the execution result of a block. It retrieves the execution result for the specified block ID.
+// Expected errors:
+// - codes.Internal if an internal error occurs while retrieving execution result.
+func (b *backendSubscribeTransactions) searchForExecutionResult(
+	blockID flow.Identifier,
+) (bool, error) {
+	result, err := b.executionResults.ByBlockID(blockID)
+	if err != nil {
+		if !errors.Is(err, storage.ErrNotFound) {
+			return false, rpc.ConvertError(err, fmt.Sprintf("failed to get execution result for block %s", blockID), codes.Internal)
```

Comment:
`ConvertError` really only handles context errors, which we won't have here. you can just directly set the internal error
```suggestion
			return false, status.Errorf(codes.Internal, "failed to get execution result for block %s: %v", blockID, err)
```